### ISSUE TITLE
Implement $nearSphere matching in the local query engine

### ISSUE THREAD

**Issue by chauthai** [ParsePlatform/ParseReact#82]
Hi,

I have this query in my react native app:

```javascript
observe(props, state) {
  return{
    events: (new Parse.Query('Event')).near('location', this.state.location)
  };
},
```

When I add a new Event object only the first newly created object will be pushed to the subscription.
Every subsequent addition will be ignored. If I do the query without the near condition the subscription works.

```javascript
observe(props, state) {
  return{
    events: (new Parse.Query('Event'))
  };
},
```

**Comment by andrewimm** [comment]
`near` matching has not been implemented in the local query engine, but it's something we can probably easily add.

Also, it's unrelated, but `this.state` will refer to the state before the observe executes, while the state parameter in the method will refer to the next state in the event of a state change. Is this what you intended to do?

**Comment by chauthai** [comment]
It would be really nice to have `near` matching!

And thanks for pointing out the `this.state` usage. What I wanted to do was to query new events when the users position changes. So I guess I should just use the `state` parameter.

**Comment by andrewimm** [comment]
Committed in cdea95a9f6af4fc06b9c4c45b5310203e6f4c83b
I want to do some more testing before releasing a new version, but it should be coming shortly.

**Comment by andrewimm** [comment]
Currently adding a case for $within, so that `.withinGeoBox()` queries can update as well

**Comment by andrewimm** [comment]
Landed in 0.4.2 last week
